builder/azure/arm/step_get_ip_address.go

```diff
@@ -11,52 +11,86 @@ import (
 	"github.com/mitchellh/packer/packer"
 )
 
+type NetworkCommunicationType int
```

Comment:
I have an itch about 'Network' in these naming scheme's. The 'network' is not different in either case, it's still a vnet, but the endpoint to which packer connects is different (public IP vs nic IP or VIP vs DIP)...